### PR TITLE
Export Connections to CSV File

### PR DESCRIPTION
**Pull request by stevenelleman**
Also splits off import deletion into a separate section for destructive settings.

### CODE REVIEW

File: apps/frontend/src/lib/exports/index.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    displayName: userData.displayName,
+    bio: userData.bio,
+    signaturePublicKey: userData.signaturePublicKey,
+    encryptionPublicKey: userData.encryptionPublicKey,
+    psiPublicKeyLink: userData.psiPublicKeyLink,
+    twitterHandle: userData.twitter?.username,
+    signalHandle: userData.signal?.username,
+    instagramHandle: userData.instagram?.username,
+    farcasterHandle: userData.farcaster?.username,
+    pronouns: userData.pronouns,
```
Comment (stevenelleman) [inline apps/frontend/src/lib/exports/index.ts:20]: Not all the UserData attributes are included.

File: apps/frontend/src/lib/exports/index.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    '\n' +
+    connections.map(connection => {
+      return keys.map(key => {
+        const keyTyped = key as keyof typeof connection;
+        const cell = connection[keyTyped] === undefined ? '' : connection[keyTyped];
+        return cell;
+      }).join(separator);
+    }).join('\n');
+
+  return new Blob([csvContent], { type: 'text/csv;charset=utf-8;' });
```
Comment (stevenelleman) [inline apps/frontend/src/lib/exports/index.ts:38]: Source: https://gist.github.com/changhuixu/de092ee55a9e115abba988910bd68d41

File: apps/frontend/src/lib/exports/index.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+      // Create a URL for the Blob
+      const url = URL.createObjectURL(blob);
+
+      // Create an anchor tag for downloading
+      const a = document.createElement('a');
+
+      // Set the URL and download attribute of the anchor tag
+      a.href = url;
+      a.download = 'cursive_connections.csv';
```
Comment (stevenelleman) [inline apps/frontend/src/lib/exports/index.ts:81]: Probably should make this a constant and put it with the other constants.

File: apps/frontend/src/pages/profile/index.tsx [hunk 4]
```diff
@@ -393,6 +400,24 @@ const ProfilePage: React.FC = () => {
             </div>
           </div>
         </div>
+
+        <div className="p-4">
+          <div className="flex flex-col gap-3">
+            <div className="flex flex-col gap-1">
+              <span className="text-lg font-semibold text-label-primary font-sans">
+                Scary Settings
```
Comment (stevenelleman) [inline apps/frontend/src/pages/profile/index.tsx:408]: Could use a better name -- "Destructive Settings"?

**Comment by stevenelleman** [comment]
Current UI: 
image

Design: 
image

Justifications for differences: 
- Move import deletion to its own section to prevent accidental deletion 
- Label with "connections" rather than "data" to make it explicit what will be exported.

File: apps/frontend/src/pages/people/index.tsx [hunk 5]
```diff
@@ -209,8 +216,18 @@ const PeoplePage: React.FC = () => {
       seoTitle="People"
       header={
         <div className="flex flex-col">
-          <span className="text-label-primary text-xl leading-none font-bold tracking-[-0.1px] py-4">
+          <span className="text-label-primary text-xl leading-none font-bold tracking-[-0.1px] py-4 inline-flex">
             {`Connections (${Object.keys(connections)?.length})`}
+            <div className="justify-start items-center gap-[40-px] flex">
```
Comment (stevenelleman) [inline apps/frontend/src/pages/people/index.tsx:221]: Current: 
image

Designs: 
image

The export button works but I haven't figured out the styling to right-align the button.

Comment (stevenelleman) [reply apps/frontend/src/pages/people/index.tsx:221]: image

File: apps/frontend/src/lib/exports/index.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import { toast } from "sonner";
+
+function flattenUserData(userData: UserData): FlattenedUserData {
+  return {
+    username: userData.username,
+    displayName: userData.displayName,
+    bio: userData.bio,
+    signaturePublicKey: userData.signaturePublicKey,
+    encryptionPublicKey: userData.encryptionPublicKey,
+    psiPublicKeyLink: userData.psiPublicKeyLink,
```
Comment (AndrewCLu) [inline apps/frontend/src/lib/exports/index.ts:15]: don't need `psiPublicKeyLink`, this isn't used anymore

File: apps/frontend/src/lib/storage/types/user/userData/index.ts [hunk 7]
```diff
@@ -44,6 +44,23 @@ export const UnregisteredUserDataSchema = z.object({
 
 export type UnregisteredUserData = z.infer<typeof UnregisteredUserDataSchema>;
 
+export const FlattenedUserDataSchema = z.object({
+  username: z.string(),
+  displayName: z.string(),
+  bio: z.string(),
+  signaturePublicKey: z.string(),
+  encryptionPublicKey: z.string(),
+  psiPublicKeyLink: z.string().optional(),
```
Comment (AndrewCLu) [inline apps/frontend/src/lib/storage/types/user/userData/index.ts:53]: don't need `psiPublicKeyLink`, same as above

**Review by AndrewCLu — APPROVED** [review]
nice, tested and works

**Pull request merged**